pkg/machines/libvirt-dbus.es6

```diff
@@ -315,6 +317,47 @@ LIBVIRT_DBUS_PROVIDER = {
         };
     },
 
+    DETACH_DISK({
+        name,
+        connectionName,
+        id: vmPath,
+        target,
+        live
+    }) {
+        /*
+         * 0 -> VIR_DOMAIN_AFFECT_CURRENT
+         * 1 -> VIR_DOMAIN_AFFECT_LIVE
+         * 2 -> VIR_DOMAIN_AFFECT_CONFIG
+         */
+        let detachFlags = 0;
+        if (live)
+            detachFlags |= 1;
+
+        return dispatch => {
+            clientLibvirt[connectionName].call(vmPath, 'org.libvirt.Domain', 'GetXMLDesc', [0], TIMEOUT)
+                    .done(domXml => {
```

Comment:
Since we talked about that yesterday: Let's please use `.then` and `.catch()` instead of the old-style jQuery `.done` and `.fail()` in new code. I'm fine with this being a separate commit or PR, of course.

Reply:
I have it consistently done for cockpit dbus client in this file. I 'll post a separate PR changing this in the whole file.

Reply:
Right, that's what I meant. Thanks!